core/parentchain/block-importer/src/block_importer.rs

```diff
@@ -176,11 +190,34 @@ impl<
 				},
 				Err(_) => error!("Error executing relevant extrinsics"),
 			};
-			let queue_game: Option<u64> = self
+			let queue_game: Option<Queue<H256>> = self
 				.ocall_api
 				.get_storage_verified(RegistryStorage::queue_game(), block.header())?
 				.into_tuple()
 				.1;
+			match queue_game {
+				Some(_) => {
+					error!("YEAHHHHH! FOUND QUEUE GAME!!!!!!!!! ");
+					let game_engine = GameEngine::new(1u8, 1u8);
+					let games = Vec::<H256>::new();
+					let shards = self.file_state_handler.list_shards().unwrap();
+					let opaque_call =
+						OpaqueCall::from_tuple(&([GAME_REGISTRY_MODULE, ACK_GAME], &game_engine, games, shards[0]));
```

Comment:
Can you add a `//FIXME` comment here as well,  that we currently only take the first shard, because we need to discuss first, how we handle sharding in general?